doc/mgr/ansible.rst

```diff
@@ -41,25 +41,32 @@ Enable the Ansible orchestrator module and use it with the :ref:`CLI <orchestrat
 Configuration
 =============
 
+The external Ansible Runner Service uses TLS mutual authentication to allow clients to use the API.
+A client certificate and a key files should be provided by the Administrator of the Ansible Runner Service for each manager node.
+This files should be copied in each of the manager nodes with read access for the ceph user.
+The destination folder for this files and the name of the files must be the same always in all the manager nodes,
+althought the certificate/key content of this files logically will be different in each node.
+
 Configuration must be set once the module is enabled by first time.
 
 This can be done in one monitor node via the configuration key facility on a
 cluster-wide level (so they apply to all manager instances) as follows::
 
 
-    # ceph config set mgr mgr/ansible/server_addr <ip_address/server_name>
-    # ceph config set mgr mgr/ansible/server_port <port>
-    # ceph config set mgr mgr/ansible/username <username>
-    # ceph config set mgr mgr/ansible/password <password>
-    # ceph config set mgr mgr/ansible/verify_server <verify_server_value>
+    # ceph config set mgr mgr/ansible/server_url <ip_address/server_name>:<port>
+    # ceph config set mgr mgr/ansible/client_cert <path_to_client_certificate_file>
```

Comment:
This is going to be a huge burden for users. Please provide a way to store the blob using `set_store` similar to `ssh.module.SSHOrchestrator#_set_ssh_config`

Reply:
I do not understand why it is going to be a burden. To use the Ansible Runner Service the user must copy the client certificate and key files in the manager host. 
Once the certificate files are in the manager, the Ansible orchestrator makes reference to them and this is all ... 

This forces to use the same destination path for the certificates in all the managers, but i think that this is a good practice.
A ssh config file probably gives more flexibility but in this case, these certificates are only for mgr hosts, and i think  is wise and healthy that "rules" will be the same for all the mgrs.

Reply:
There is a tendency to move thins from physical config files into the ceph cluster: 

1. The ganesha config is nowadays stored inside of the cluster and no longer maintained externally,
2. The SSL certificates for the dashboard and the prometheus module are now stored in the cluster
3. She ssh config of the ssh orchestrator is stored in the cluster,

This has some real benefits: Ceph itself automatically takes cares of making the config available and thus reduces the amount of steps needed to have a valid configuration: It simply reduces the number of things that can go wrong, which is a good thing.

On the other hand, you're of course right: The ansible orchestrator only needs to run in clusters deployed by ansible. I'd still recommend to use `set_store`, but I'm not blocking this PR

Reply:
ok, Thank you for your explanation.  I see the point... i would like to  discuss this with other interested people and then come back with a decision.

Reply:
Just curious - is this a single client cert that all mgrs would use or does each mgr have a separate client cert to access the ARS (i.e. hostname is part of the cert)? 

Bottom line is I like Sebastian's suggestion to use the mon store.

@jmolmo What would the process be to populate into the store in a container context?

Reply:
@pcuzner:
You must use different certificates for each different client because the SSL Cert is bound to the actual host name.  Also you can use the same certificate for several clients if it is a wildcard SSL certificate ( generated for be used in several hosts (ex. domain= *.company)
In the current implementation in the ansible orchestrator certificates file names must be the same for all the managers,  but this does not imply to use the same certificate.( although as i said it depends of how the certificates are generated, for example, the test client certificates generated in Ansible Runner Service are wildcard SSL certs.)

@sebastian-philipp , @pcuzner :
To store the certs in the mon store has implications in how we configure the module that i would like to share:

How do we introduce introduce the certificate in the mon store? 
We can add methods to add/remove certificates/keys in the ansible module ( as is done in `ssh.module.SSHOrchestrator#_set_ssh_config`  for example) but this means that configuration of the ansible module will need to be done using another module or script.

If we change the way to provide the certificates using this new methods,  we lose the possibility of use the Ansible Orchestrator through the orchestrator-cli,  ( and this implies no command line for ansible orchestrator).
Unless it will be acceptable to introduce the needed commands in the orchestrator-cli interface. (although i think it does not seem sensible to introduce end-points for only one orchestrator)

Another option is to rely in the final user this task, let him to use "ceph config-key" commands to do the job , and after that enable the ansible module. ( i do not like very much this option... i think is better what we have now, (the settings for pointing where are the cert and key files))

I agree that to store the certs in the cluster can serve to avoid configuration problems and make things easy when adding new manager nodes for example, but i think that to lose the command line is a very high price.

I am opened to other suggestions.

Reply:
> @sebastian-philipp , @pcuzner :
> To store the certs in the mon store has implications in how we configure the module that i would like to share:
> 
> How do we introduce introduce the certificate in the mon store?
> We can add methods to add/remove certificates/keys in the ansible module ( as is done in `ssh.module.SSHOrchestrator#_set_ssh_config` for example) but this means that configuration of the ansible module will need to be done using another module or script.
> 
> If we change the way to provide the certificates using this new methods, we lose the possibility of use the Ansible Orchestrator through the orchestrator-cli, ( and this implies no command line for ansible orchestrator).

Actually, that's exactly the place I would expect this to be located. just create a new command `ceph ansible set client-certificate <host> -i <file>` 

> Unless it will be acceptable to introduce the needed commands in the orchestrator-cli interface. (although i think it does not seem sensible to introduce end-points for only one orchestrator)

I don't think we need this in the orchestrator_cli. as it is something specific for ansible.

> 
> Another option is to rely in the final user this task, let him to use "ceph config-key" commands to do the job , and after that enable the ansible module. ( i do not like very much this option... i think is better what we have now, (the settings for pointing where are the cert and key files))

There is indeed an unresolved race between the mgr startup and ceph-ansible to configure the ansible module. 

> 
> I agree that to store the certs in the cluster can serve to avoid configuration problems and make things easy when adding new manager nodes for example, but i think that to lose the command line is a very high price.

:+1:, but as I said, I don't think we're loosing the cli. 

> 
> I am opened to other suggestions.

Reply:
> I agree with @sebastian-philipp: that's more or less the same approach taken in ceph-dashboard:
> 
> ```shell
> ceph config-key set mgr/dashboard/$name/crt -i dashboard.crt
> ceph config-key set mgr/dashboard/$name/key -i dashboard.key
> ```

Done... but i have used:

```
ceph ansible set-ssl-certificate-key <host> -i /<file>
ceph ansible set-ssl-certificate <host> -i /<file>
```

To be aligned with dashboard